src/NServiceBus.Testing/Handler.cs

```diff
@@ -4,24 +4,25 @@
     using System.Collections.Generic;
     using System.Linq;
     using System.Reflection;
+    using NServiceBus.MessageInterfaces.MessageMapper.Reflection;
+    using NServiceBus.Testing.ExpectedInvocations;
 
     /// <summary>
     /// Message handler unit testing framework.
     /// </summary>
     public class Handler<T>
     {
-        private readonly StubBus bus;
         private readonly T handler;
-        private IDictionary<string, string> incomingHeaders = new Dictionary<string, string>();
-        private IList<IExpectedInvocation> expectedInvocations = new List<IExpectedInvocation>();
+        IMessageCreator messageCreator = new MessageMapper();
```

Comment:
@mat-mcloughlin @timbussmann I wanted to keep *some* inversion here, so that the `TestableMessageHandlerContext` is highly composable in testing. But, I don't mind inlining it, if that helps.